### PR TITLE
AREA-22 - Add JWT classes for token generation

### PR DESCRIPTION
**Pull request by TekMath**
I added a configuration for JWT and a jwt generator

### CODE REVIEW

File: backend/Zeus.Api.Application/Common/Interfaces/Authentication/IJwtGenerator.cs [hunk 1]
```diff
@@ -0,0 +1,8 @@
+using Zeus.Api.Domain.UserAggregate;
+
+namespace Zeus.Api.Application.Common.Interfaces.Authentication;
+
+public interface IJwtGenerator
+{
+    string Generate(User user);
```
Comment (flavien-chenu) [inline backend/Zeus.Api.Application/Common/Interfaces/Authentication/IJwtGenerator.cs:7]: Maybe you could make a difference between `GenerateAccessToken` and `GenerateRefreshToken` to be sure that a refresh can't be used an access and an access as a refresh ?

**Review by flavien-chenu — CHANGES_REQUESTED** [review]

File: backend/Zeus.Api.Domain/Common/ValueObjects/Authentication/AccessToken.cs [hunk 2]
```diff
@@ -0,0 +1,8 @@
+namespace Zeus.Api.Domain.Common.ValueObjects.Authentication;
```
Comment (flavien-chenu) [inline backend/Zeus.Api.Domain/Common/ValueObjects/Authentication/AccessToken.cs:1]: Why put it in `Common/ValueObjects/Authentication` ? I think `Authentication/ValueObjects`  should be more adapted ?

File: backend/Zeus.Api.Infrastructure/Authentication/JwtGenerator.cs [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+            expires: _dateTimeProvider.UtcNow.AddMinutes(_jwtSettings.ExpiryMinutes),
+            claims: claims,
+            signingCredentials: signingCredentials);
+
+        return new JwtSecurityTokenHandler().WriteToken(securityToken);
+    }
+
+    public AccessToken GenerateAccessToken(User user)
+    {
+        return new AccessToken(GenerateToken(user));
```
Comment (flavien-chenu) [inline backend/Zeus.Api.Infrastructure/Authentication/JwtGenerator.cs:52]: I think you should add a `Token.Type` in the claims to make difference between an `AccessToken` and a `RefreshToken` to avoid use one for the usage of an other. So, if you do it, you'll have to check at decoding time the type of each token.

**Review by flavien-chenu — CHANGES_REQUESTED** [review]

**Review by flavien-chenu — CHANGES_REQUESTED** [review]
I think encapsulating the token strings in classes isn’t really necessary in this case, but why not—it just doesn’t add much value in my opinion.

Here’s what I propose:

- No more separate classes for tokens.
- Instead, you have a static `TokenType` class that contains the token strings, like `TokenType.AccessToken = nameof(AccessToken)`...
- You pass a user and a token type to the `GenerateToken` method, which maps the user’s properties into claims and adds the token type as well.

In short, I don’t think it’s very relevant to encapsulate the JWTs into classes like `Token`, `AccessToken`, etc., given that we know a JWT is always a string—it’s a standard. Moreover, the classes are essentially empty.

We just need to verify the token type during decoding to ensure it matches `TokenType.AccessToken` or `TokenType.Refresh`.

File: backend/Zeus.Api.Domain/Authentication/ValueObjects/Token.cs [hunk 4]
```diff
@@ -0,0 +1,18 @@
+using Zeus.Api.Domain.Common.Models;
+
+namespace Zeus.Api.Domain.Common.ValueObjects.Authentication;
+
+public class Token : ValueObject
```
[the remark this answers is not in the document]
Comment (TekMath) [reply backend/Zeus.Api.Domain/Authentication/ValueObjects/Token.cs:5]: No because you can create a simple Token for a no-specific usage

File: backend/Zeus.Api.Infrastructure/Authentication/JwtGenerator.cs [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+            new SymmetricSecurityKey(
+                Encoding.UTF8.GetBytes(_jwtSettings.Secret)),
+            SecurityAlgorithms.HmacSha256);
+        var claims = new[]
+        {
+            new Claim(JwtRegisteredClaimNames.Sub, user.Id.Value.ToString()),
+            new Claim(JwtRegisteredClaimNames.GivenName, user.FirstName),
+            new Claim(JwtRegisteredClaimNames.FamilyName, user.LastName),
+            new Claim(JwtRegisteredClaimNames.Jti, Guid.NewGuid().ToString()),
+            new Claim(JwtRegisteredCustomClaimNames.TokenType, type)
```
[the remark this answers is not in the document]
Comment (TekMath) [reply backend/Zeus.Api.Infrastructure/Authentication/JwtGenerator.cs:38]: NO! Typ is a native property used to say if your token is in JWT format. It's prohibited to modified this value

**Review by flavien-chenu — APPROVED** [review]
